### PR TITLE
fix(TDI-36871): Validate json field value on write in JIRA output com…

### PR DESCRIPTION
**Pull request by iistomin-gl**
…ponent

**What is the current behavior?** (You can also link to an open issue here)
* JIRA output component doesn't check json field for null/empty value

**What is the new behavior?**
* JIRA output component checks json field for null/empty value before submitting of a request

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [X] The code coverage on new code >75%
- [X] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

**Other information**:
JIRA issue: https://jira.talendforge.org/browse/TDI-36871

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/823/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/823/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/824/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/824/)

File: components/components-jira/src/main/java/org/talend/components/jira/runtime/writer/JiraWriter.java [hunk 1]
```diff
@@ -197,4 +199,17 @@ protected IOException createRejectException(String reasonKey, String json, Strin
         sb.append(serverMessage);
         return new IOException(sb.toString());
     }
+
+    /**
+     * Validate request body (json content) before submitting of a request.
+     *
```
Comment (gonchar-ivan) [inline components/components-jira/src/main/java/org/talend/components/jira/runtime/writer/JiraWriter.java]: It is worth to mention that it checks for null and empty value

File: components/components-jira/src/main/java/org/talend/components/jira/runtime/writer/JiraInsertWriter.java [hunk 2]
```diff
@@ -79,7 +80,13 @@ public void write(Object datum) throws IOException {
             }
             jsonPos = jsonField.pos();
         }
+
         String json = (String) record.get(jsonPos);
+        try {
+            validateRequestBody(json);
+        } catch (DataRejectException e) {
+            throw createRejectException("error.invalidRecordCreate", json, (String) e.getRejectInfo().get("message"));
+        }
```
Comment (gonchar-ivan) [inline components/components-jira/src/main/java/org/talend/components/jira/runtime/writer/JiraInsertWriter.java]: This is the same as in JiraUpdateWriter. I suggest to put Reject Exception creation to validateRequestBody() method.
Here, you may leave only a call to validateRequestBody(json);

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/858/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/858/)

File: components/components-jira/src/main/java/org/talend/components/jira/runtime/writer/JiraWriter.java (hunk 1 above)
Comment (chmyga) [inline components/components-jira/src/main/java/org/talend/components/jira/runtime/writer/JiraWriter.java]: totally agree.

**Pull request merged**